### DISCUSSION TITLE
Inquietud LAB Numpy - Task 3

### DISCUSSION THREAD

**Discussion by StevieCL** [rramosp/ai4eng.v1#11] [02 - PYTHON]
Hola, buenas noches. Tengo el siguiente problema en el desarrollo de la task 3; en la prueba manual, el código me funciona, pero cuando lo voy a subir me saca un problema. No sé realmente a qué se refiere el error: no sé si el error será cuando el programa calcula la transpuesta o la multiplicación entre el vector de promedios y el array de ones. Dejo adjuntas imágenes de mi código respuesta y lo que me saca el submit. 
![Screenshot from 2021-05-17 01-59-41](https://user-images.githubusercontent.com/76596773/118445829-87952880-b6b4-11eb-97c5-d027e6093b7b.png)
![Screenshot from 2021-05-17 01-59-47](https://user-images.githubusercontent.com/76596773/118445854-8d8b0980-b6b4-11eb-88b8-973adda082d8.png)

**Comment by rramosp** [comment]
hola Stiven ... tienes que tener en cuenta que a lo mejor la matriz de entrada no es cuadrada por lo que  el mean con axis=1 puede que no te coincida con np.ones ... 

la solución de usar np.ones es ingeniosa ... te sugiero que uses un .reshape al resultado del np.mean para darle la forma que necesitas en el caso de matrices no cuadradas.

espero así logres que te funcione. nos dices si no es así

saludos
raúl

**Reply by StevieCL** [reply]
Hola profe, muchas gracias! Ya pude realizar el punto.

**Reply by rramosp** [reply]
perfecto!!
